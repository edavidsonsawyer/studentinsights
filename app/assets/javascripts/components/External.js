import PropTypes from 'prop-types';
import React from 'react';


// External link
export default function External(props) {
  const {children} = props;
  return (
    <a
      className="External"
      target="_blank"
      rel="noopener noreferrer"
      {...props}
    >{children}</a>
  );
}
External.propTypes = {
  children: PropTypes.node.isRequired
};
