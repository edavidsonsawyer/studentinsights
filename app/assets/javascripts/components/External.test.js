import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import External from './External';

function testProps(props) {
  return {
    href: 'https://example.com/hello',
    children: 'example.com',
    ...props
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(<External {...testProps()} />, el);
  expect(el.innerHTML).toContain('hello');
});

it('snapshots view', () => {
  const tree = renderer
    .create(<External {...testProps()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});