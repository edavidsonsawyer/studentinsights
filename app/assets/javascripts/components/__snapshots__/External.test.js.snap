// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots view 1`] = `
<a
  className="External"
  href="https://example.com/hello"
  rel="noopener noreferrer"
  target="_blank"
>
  example.com
</a>
`;
